### DISCUSSION TITLE
Capacitor Wish List

### DISCUSSION THREAD

**Discussion by MinimogDev** [ionic-team/capacitor#3298] [Ideas]
Hey everyone!

I hope discussions are a right place to post something like this and I think timing might be right for this while v3 is being planned https://github.com/ionic-team/capacitor/issues/3133

I wanted to compile a list of "wishes" that I am personally currently missing from capacitor. I think some of these might be duplicates of what has been posted while others are new.

#### 1. Official guide on managing different iOS Schemas and Android Flavours (Development, Staging, Production etc...
Coming from web dev world most of us are used to setting these up with different config files and such, but in native land I found it to be a tad more involved and a nice guide on this would be a good idea to add to the docs imho. Some thing that showcases how to create different ios schemas and android flavours and how to use different assets for app icons / configure different app names and bundle id's etc...

#### 2. Support for env based capacitor config files
To my understanding capacitor only supports 1 `capacitor.config.js` file per project at the moment. I think it would be good to allow us define multiple based on needs of our enviroments i.e. `{env}.capacitor.config.js`. Being able to set things up in files as opposed to cli for different envs would be a massive help.

#### 3. Better experience around splash screen customization
 I saw several issues related to splash screen customisation and struggled with this myself as well, it would be good to see a guide showcasing how to customise these (on a native level / as native views) and ensure they are responsive on various screen sizes.

#### 4. Better experience around upgrading native projects
Right now we need to keep track of native project changes and go through various commits and apply these changes. It would be great to see a tool like https://react-native-community.github.io/upgrade-helper/?from=0.62.2&to=0.63.1 for capacitor's native projects integrated into main site alongside release notes.

#### 5. Plugin Hub
Some projects like flutter have separate hubs / pages for plugins i.e. https://pub.dev/packages?q=firebase that allow to search them and show their health / stats alongside these search results. With capacitor something like this would be cool to see how popular and well maintained a plugin / does it support typescript and other useful stats

#### 6. Documentation Search
Self explanatory

**Comment by imhoffd** [comment]
Thanks @MinimogDev, this is great! 

1. Great idea. We want to investigate this more and determine if it makes sense to try accomplishing this with tooling or if documentation would be a better approach. Supporting multiple environments is a priority for Capacitor 3. I just created https://github.com/ionic-team/capacitor/issues/3303 to track this.
2. Yes! This is listed in the Capacitor 3 planning issue under "Dynamic Capacitor Config".
3. Could you go into this a little more? What sort of tooling might you expect from Capacitor?
4. It looks like that tool shows the differences in newly generated React Native projects, using [this repo](https://github.com/react-native-community/rn-diff-purge). This is really neat! And, it'll be easy once we implement [automated releases](https://github.com/ionic-team/capacitor/issues/3289). Capacitor is a lot younger than React Native, but with any luck we'll have time to implement this. I created https://github.com/ionic-team/capacitor/issues/3302 if you'd like to follow it.
5. One initiative for Capacitor 3 is more community involvement. We've been actively bringing people together to create the [Capacitor Community GitHub org](https://github.com/capacitor-community/) for this very purpose. Read more [here](https://github.com/capacitor-community/welcome#welcome).
6. This is being worked on right now! You can follow this issue for progress: https://github.com/ionic-team/capacitor-site/issues/64

**Comment by aparajita** [comment]
> 3. Better experience around splash screen customization

@MinimogDev @dwieeb I've got this covered! New splash screen plugin coming within the next week or so allows complete customization of the splash with all possible native options and support for adding your own custom animation.
